src/Grpc.Net.Client/Internal/GrpcProtocolHelpers.cs

```diff
@@ -252,38 +273,39 @@ internal static AuthInterceptorContext CreateAuthInterceptorContext(Uri baseAddr
             serviceUrl += "/";
         }
         serviceUrl += method.ServiceName;
-        return new AuthInterceptorContext(serviceUrl, method.Name);
+        return new AuthInterceptorContext(serviceUrl, method.Name, cancellationToken);
     }
 
     internal static async Task ReadCredentialMetadata(
         DefaultCallCredentialsConfigurator configurator,
         GrpcChannel channel,
         HttpRequestMessage message,
         IMethod method,
-        CallCredentials credentials)
+        CallCredentials credentials,
+        CancellationToken cancellationToken)
     {
         credentials.InternalPopulateConfiguration(configurator, null);
 
         if (configurator.Interceptor != null)
         {
-            var authInterceptorContext = GrpcProtocolHelpers.CreateAuthInterceptorContext(channel.Address, method);
+            configurator.CachedContext ??= CreateAuthInterceptorContext(channel.Address, method, cancellationToken);
```

Comment:
This looked scary at first, but the call sites don't reuse the `DefaultCallCredentialsConfigurator` for different channel + method + cancellation combos.

Can we make that more obvious somehow so this code doesn't look scary?

e.g.

```
ReadCredentialMetadata(...)
{
    ReadCredentialMetadataCore(...);
    configurator.CachedContext = null;
}

// Rename current method
ReadCredentialMetadataInner(...)
{
}
```

Reply:
I've updated comments and renamed some fields to make it clearer what is going on.